riscv-page-tables/src/page_table.rs

```diff
@@ -585,6 +600,64 @@ impl<T: PagingMode> PageTableInner<T> {
         entry
     }
 
+    /// Creates contiguous mappings, both physically and virtually, of `num_pages` pages, starting
+    /// from `start_vaddr` and `start_paddr`, with the same permissions.
+    ///
+    /// # Safety
+    ///
+    /// The caller must guarantee that app pages reference from `start_paddr` for `num_pages` are
+    /// uniquely owned by the root `GuestStagePageTable`.
+    #[allow(dead_code)]
+    pub unsafe fn map_contiguous_leaves(
+        &mut self,
+        start_vaddr: PageAddr<T::MappedAddressSpace>,
+        start_paddr: SupervisorPageAddr,
+        num_pages: u64,
+        page_size: PageSize,
+        perms: PteFieldBits,
+    ) -> Result<()> {
+        start_vaddr
+            .checked_add_pages_with_size(num_pages, page_size)
+            .ok_or(Error::OutOfMapRange)?;
```

Comment:
nit: we can't really say this is because it's out of map range. It's more checking if the address overflowed over `u64`, or if the address wasn't aligned with the page size.

Reply:
Heh. Well one could say that if the end of the range overflows the address space we're out of range. :-)

That was essentially the error passed before. Speficifying AddressOverflow over OutOfMap would gather no real advantage.

Reply:
Changed to Address overflow since I was there.